### PR TITLE
Move to SendGrid

### PR DESCRIPTION
**Pull request by brianlovin**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- athena
- hermes

@mxstbr a couple things to pay attention to in the review here:
- `hermes/send-email.js` - just double check this code change please
- I wrote some scripts to automate:
  - inlining css (sendgrid does not do this automatically for us)
  - syncing all template files to sendgrid via the api - makes it easy to make sure sendgrid always has our latest template html
  - sending test emails to a person without having to use the web app to try and trigger different events

Feel free to review those automation scripts, although I imagine I'll be on the only person touching those for the foreseeable future :P 

Afaik everything else in here was fixing up templates to behave better across email clients, fixed some consistency problems, fixed some content escaping/encoding, etc.

### CODE REVIEW

**Comment by brianlovin** [comment]
@mxstbr do you know how to get Danger to ignore the fact that my new `email-template-scripts` files don't have flow enabled? Those scripts aren't run through babel so I can't use es6 import/export or flowtypes.

File: hermes/send-email.js [hunk 1]
```diff
@@ -61,55 +74,42 @@ const sendEmail = (options: Options) => {
 
   // qq.com email addresses are isp blocked, which raises our error rate
   // on postmark. prevent sending these emails at all
```
Comment (mxstbr) [inline hermes/send-email.js]: nit: postmark => sendgrid

**Review by mxstbr — DISMISSED** [review]
- Took a short look at `send-email`, looks fine to me!
- Is there a particular reason you wrote the script with python? Would have preferred to just stick with JavaScript since I don't know python and can't debug in a worse case scenario… I guess it's not worth rewriting now but next time I'd prefer if it was in JS 👍 
- Love that we automatically sync templates now, that was dearly missing from Postmark! 😍 
- Amazing tests 👏 
- Escaping 👏 👏 👏 👏 👏 

Thanks for slogging through this, much appreciated :pray: Still sad we can't stick with Postmark, I really liked them 😢 

LGTM overall

**Comment by brianlovin** [comment]
> Is there a particular reason you wrote the script with python? Would have preferred to just stick with JavaScript since I don't know python and can't debug in a worse case scenario… I guess it's not worth rewriting now but next time I'd prefer if it was in JS 👍

Original lib is python, but I just found https://github.com/JedWatson/node-premailer - I can see if it's a quick change so we stay 100% in js land.

> Still sad we can't stick with Postmark, I really liked them

Same 😢

**Comment by brianlovin** [comment]
> Original lib is python, but I just found https://github.com/JedWatson/node-premailer - I can see if it's a quick change so we stay 100% in js land.

Actually, I'm just going to leave it python if that's okay. Jed's actually is sending API requests to the premailer website and awaits an API response. I'd like to not have that dependency, esp since the python script just...works right now. If we ever need to dig deeper than what it's already doing I can investigate a switch :)

**Comment by mxstbr** [comment]
Okay!

**Comment by brianlovin** [comment]
@mxstbr you probably won't code review this before we have to ship today, but those last 3 commits add the optional request handler to `createWorker` - it's not clean, but it works. Open to feedback, although it's possible this will be live when you wake up (of course, I tested everything locally - ngrok to the rescue!)

**Review by mxstbr — DISMISSED** [review]
LGTM, this is a fine escape hatch for now!

**Pull request merged**